heat/sparse/manipulations.py

```diff
@@ -2,16 +2,51 @@
 from __future__ import annotations
 
 from heat.sparse.dcsr_matrix import DCSR_matrix
-
+from heat.sparse.factories import sparse_csr_matrix
 from ..core.memory import sanitize_memory_layout
 from ..core.dndarray import DNDarray
 from ..core.factories import empty
 
 __all__ = [
     "todense",
+    "to_sparse",
 ]
 
 
+def to_sparse(array: DNDarray) -> DCSR_matrix:
+    """
+    Convert the distributed array to a sparse DCSR_matrix representation.
+
+    Parameters
+    ----------
+    array : DNDarray
+        The distributed array to be converted to a sparse DCSR_matrix.
+
+    Returns
+    -------
+    DCSR_matrix
+        A sparse DCSR_matrix representation of the input DNDarray.
+
+    Notes
+    -----
+    This method allows for the conversion of a DNDarray into a sparse DCSR_matrix representation,
+    which is useful for handling large and sparse datasets efficiently.
+
+    Examples
+    --------
+    >>> dense_array = ht.array([[1, 0, 0], [0, 0, 2], [0, 3, 0]])
+    >>> sparse_matrix = dense_array.to_sparse()
+
+    """
+    array.balance_()
+    result = sparse_csr_matrix(array.larray, is_split=array.split)
```

Comment:
Hey @Mystic-Slice, I'm not sure but calling sparse_csr_matrix with only an array is giving the following error.
```
Traceback (most recent call last):
  File "/media/samadpls/ubuntu/ML/heat/test.py", line 7, in <module>
    B = A.to_sparse()
  File "/media/samadpls/ubuntu/ML/heat/heat/sparse/manipulations.py", line 42, in to_sparse
    result = sparse_csr_matrix(array, dtype=array.dtype, device=array.device, comm=array.comm, is_split=array.split)
  File "/media/samadpls/ubuntu/ML/heat/heat/sparse/factories.py", line 120, in sparse_csr_matrix
    obj = torch.tensor(
  File "/media/samadpls/ubuntu/ML/heat/heat/core/dndarray.py", line 1116, in __len__
    return self.shape[0]
IndexError: tuple index out of range
```
I checked other test cases and found that sparse_csr_matrix is called with `torch.Tensor` rather than DNDarray, as in this case.

Reply:
Alright. Let's just use `array.larray` then.